WgWall/ClientApp/src/app/products/products.component.ts

```diff
@@ -104,15 +104,14 @@ export class ProductsComponent {
   }
 
   confirmBuy(product: Product) {
-    product.boughtById = this.user.id;
     this.active.splice(this.active.indexOf(product), 1);
     this.update(product);
     this.user.karma = +this.user.karma + +product.amount;
   }
 
   hide(name: string) {
     this.productService.hideAll(name);
-    this.products.filter(p => p.name == name).forEach(p => p.hide = true);
+    this.products.filter(p => p.name == name);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The return value of "filter" must be used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS2201)